src/charm.py

```diff
@@ -354,6 +355,8 @@ def _on_config_changed(self, event: EventBase) -> None:
             )
             self.kafka_config.set_server_properties()
 
+            self.kafka_config.set_environment()
```

Comment:
I think [comment from zk PR](https://github.com/canonical/zookeeper-operator/pull/105#discussion_r1427912603) also applies here. Not a big deal though.